rag_experiment_accelerator/evaluation/plain_metrics.py

```diff
@@ -1,153 +1,163 @@
 
     Args:
-        value1 (str): The first string to compare.
-        value2 (str): The second string to compare.
+        str1 (str): The first string to compare.
+        str2 (str): The second string to compare.
 
     Returns:
         int: The normalized similarity score as a percentage.
     """
-    score = int(algorithms.levenshtein.normalized_similarity(value1, value2) * 100)
+    score = d.Levenshtein.normalized_similarity(str1, str2) * 100
     return score
 
 
-def jaccard(value1, value2):
+def jaccard(str1: str, str2: str) -> int:
     """
     Calculates the Jaccard similarity score between two sets of values.
 
     Args:
-        value1 (set): The first set of values.
-        value2 (set): The second set of values.
+        str1 (set): The first set of values.
+        str2 (set): The second set of values.
 
     Returns:
         int: The Jaccard similarity score between the two sets of values, as a percentage.
     """
-    score = int(algorithms.jaccard.normalized_similarity(value1, value2) * 100)
+    score = int(alg.jaccard.normalized_similarity(str1, str2) * 100)
     return score
 
 
-def hamming(value1, value2):
+def hamming(str1: str, str2: str) -> int:
     """
     Calculates the Hamming similarity score between two values.
 
     Args:
-        value1 (str): The first value to compare.
-        value2 (str): The second value to compare.
+        str1 (str): The first value to compare.
+        str2 (str): The second value to compare.
 
     Returns:
         int: The Hamming similarity score between the two values, as a percentage.
     """
-    score = int(algorithms.hamming.normalized_similarity(value1, value2) * 100)
+    score = int(d.Hamming.normalized_similarity(str1, str2) * 100)
     return score
 
 
-def jaro_winkler(value1, value2):
+def jaro_winkler(str1: str, str2: str) -> int:
     """
     Calculates the Jaro-Winkler similarity score between two strings.
 
     Args:
-        value1 (str): The first string to compare.
-        value2 (str): The second string to compare.
+        str1 (str): The first string to compare.
+        str2 (str): The second string to compare.
 
     Returns:
         int: The Jaro-Winkler similarity score between the two strings, as an integer between 0 and 100.
     """
-    score = int(algorithms.jaro_winkler.normalized_similarity(value1, value2) * 100)
+    score = int(d.JaroWinkler.normalized_similarity(str1, str2) * 100)
     return score
 
 
-def cosine(value1, value2):
+def cosine(str1: str, str2: str) -> int:
     """
     Calculates the cosine similarity (Ochiai coefficient) between two strings
     using token-frequency vectors
 
     https://en.wikipedia.org/wiki/Cosine_similarity.
 
     Args:
-        value1 (list): The first vector.
-        value2 (list): The second vector.
+        str1 (list): The first vector.
+        str2 (list): The second vector.
```

Comment:
removed the cosine function from this file as all other functions here take in strings or list of strings to compute simple metrics. cosine assumes there needs to be some encoding of strings to a vector space which we are doing in the transforner based metrics anyway.

Reply:
Tag @ritesh-modi

Reply:
The cosine function implemented here is not the traditional implementation used with embeddings, but rather, the Ochiai coefficient (see [here](https://en.wikipedia.org/wiki/Cosine_similarity)). It takes in two lists of strings so I don't think the renaming of the variables is misleading. I'd suggest to keep the metric and maybe update the name of the metric to avoid confusion (e.g. "cosine_ochiai")

Reply:
ah ok. will retain and rename the cosine metric then. between the signature and variable name present earlier, it was unclear what it was measuring